src/main/java/com/box/sdk/BoxTermsOfService.java

```diff
@@ -0,0 +1,298 @@
+         * @return the status of the terms of service.
+         */
+        public String getStatus() {
+            return this.status;
+        }
+
+        /**
+         * Sets the status of the terms of service in order to enable or disable it.
+         *
+         * @param status the new status of the terms of service.
+         */
+        public void setStatus(String status) {
+            this.status = status;
+            this.addPendingChange("status", status.toLowerCase());
+        }
+
+        /**
+         * TermsOfServiceType can be "managed" or "external".
+         * @return the type of the terms of service.
+         */
+        public String getType() {
+            return this.type;
+        }
+
+        /**
+         * The type is terms_of_service.
+         * @return the type terms_of_service.
+         */
+        public String getTosType() {
+            return this.tosType;
+        }
+
+
+        /**
+         * @return the enterprise for the terms of service.
+         */
+        public BoxEnterprise getEnterprise() {
+            return this.enterprise;
+        }
+
+
+        /**
+         * @return the text of the terms of service.
+         */
+        public String getText() {
+            return this.text;
+        }
+
+        /**
+         * Sets the text of the terms of service.
+         *
+         * @param text the new text of the terms of service.
+         */
+        public void setText(String text) {
+            this.text = text;
+            this.addPendingChange("text", text);
+        }
+
+
+        /**
+         * @return time the policy was created.
+         */
+        public Date getCreatedAt() {
+            return this.createdAt;
+        }
+
+        /**
+         * @return time the policy was modified.
+         */
+        public Date getModifiedAt() {
+            return this.modifiedAt;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        void parseJSONMember(JsonObject.Member member) {
+            super.parseJSONMember(member);
+            String memberName = member.getName();
+            JsonValue value = member.getValue();
+            try {
+                if (memberName.equals("status")) {
+                    this.status = value.asString();
+                } else if (memberName.equals("enterprise")) {
+                    JsonObject jsonObject = value.asObject();
+                    if (this.enterprise == null) {
+                        this.enterprise = new BoxEnterprise(jsonObject);
+                    } else {
+                        this.enterprise.update(jsonObject);
+                    }
+                } else if (memberName.equals("type")) {
+                    this.type = value.asString();
+                } else if (memberName.equals("tos_type")) {
+                    this.tosType = value.asString();
+                } else if (memberName.equals("text")) {
```

Comment:
value is mandatory field? if there is any optional fields, the null checks are done?